### ISSUE TITLE
Add intention to expand `S { .. }` in patterns

### ISSUE THREAD

**Issue by vlad20012** [intellij-rust/intellij-rust#3928]
In rust it's possible to skip some fields in patterns by adding `..`:
```rust
struct S { a: i32, b: i32, c:i32 }
fn foo(s: S) {
    let S { a, .. } = s;
}
```
And for tuple struct case:
```rust
struct S(i32, i32, i32);
fn foo(s: S) {
    let S(a, ..) = s;
}
```

It would be nice to have an intention to replace such `..` with the actual fields.

**Comment by shevtsiv** [comment]
If no objection, I would like to work on this issue.

**Comment by vlad20012** [comment]
Maybe it's reasonable to implement analysis for E0027 & E0023 first

**Comment by Undin** [comment]
Fixed in #3996
